### PR TITLE
Prac 06 code review request

### PR DESCRIPTION
**Pull request by scott-gills-dyer**
Check code structure, consistency, and variable names

@Kanos812

### CODE REVIEW

File: prac_06/car.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        fuel: float, one unit of fuel drives one kilometre
+        """
+        self.name = name
+        self.fuel = fuel
+        self._odometer = 0
+
+    def __str__(self):
+        """Return a string representation of car object."""
+        return f"{self.name}, {self.fuel}, {self._odometer}"
+
```
Comment (Kanos812) [inline prac_06/car.py:19]: Current __str__ method works, however for usability I would suggest adding in more descriptors to the strings being outputted

i.e. f"Car {self.name}, Fuel = {self.fuel}  ... "

File: prac_06/car.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    def add_fuel(self, amount):
+        """Add amount to the car's fuel."""
+        self.fuel += amount
+
```
Comment (Kanos812) [inline prac_06/car.py:23]: May be useful to add a parameter to only allow positive amounts of fuel to be added

File: prac_06/used_cars.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+from prac_06.car import Car
+
+
+def main():
+    """Demo test code to show how to use car class."""
+    my_car = Car("My Car", 180)
+    my_car.drive(30)
+    print(f"Car has fuel: {my_car.fuel}")
+    print(my_car)
+
```
Comment (Kanos812) [inline prac_06/used_cars.py:16]: Good practice to provide more commentary on lines, even if it is short 

Trying to show 'why' something has been put there

File: prac_06/programming_language.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+Predicted: 10 mins
+Actual: 20 mins
+
+"""
+
+
+class ProgrammingLanguage:
+    """Programming Language class, has name, typing, reflection, year"""
+
+    def __init__(self, name, typing, reflection, year):
```
Comment (Kanos812) [inline prac_06/programming_language.py:13]: Consider defining the output with commenting . While clear for this example may be more useful in other scripts where the result is more ambiguous (whether its a str or boolean output)

File: prac_06/programming_language.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        self.reflection = reflection
+        self.year = year
+
+    def is_dynamic(self):
+        """Determine if language is dynamic."""
+        return self.typing == "Dynamic"
+
+    def __str__(self):
+        """Return string of the language object."""
+        return f"{self.name}, {self.typing}, {self.reflection}, {self.year}"
```
Comment (Kanos812) [inline prac_06/programming_language.py:26]: Similar to cars.py I would suggest adding more descriptors to the output to increase clarity and readability

File: prac_06/guitar.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
+"""
+CP1404 Practical
+
+Estimate: 5 minutes
+Actual:   10 minutes
+
+"""
+
+CURRENT_YEAR = 2024
```
Comment (Kanos812) [inline prac_06/guitar.py:9]: I know from the instructions we were told to hardcode the year into the code, however one minor improvement would be to provide a dyamic calculation of the current year

**Review by Kanos812 — COMMENTED** [review]
Overall the code is very neat, consistent and functions well. I found it very hard to actually find issues in formatting and naming. Otherwise the code reads very easily and is great in terms of modality and the ability to refactor.

My major piece of feedback is potentially looking at integrating error handling into each of the scripts. While not necessarily required by the instructions (and from the looks of it I think you'd be more than capable of adding it in) I believe it's just a good habit to have. 

Otherwise, I don't think I found any single instance of incorrect formatting, indentation or inconsistent variable naming conventions. 

I will definitely be going back to my own code now to see how I can clean my own structure up. Sorry I cannot provide much constructive criticism however I hope it shows the quality of code that has been created (at least in my eyes, which are notoriously untrustworthy)

File: prac_06/car.py (hunk 2 above)
Comment (lindsaymarkward) [reply prac_06/car.py:23]: That's not the class's job. Can you think of any error-checking we've done in class methods?

**Pull request merged**